docs/source/cpp/json.rst

```diff
@@ -59,7 +59,7 @@ the output table.
       // Instantiate TableReader from input stream and options
       std::shared_ptr<arrow::json::TableReader> reader;
       st = arrow::json::TableReader::Make(pool, input, read_options,
```

Comment:
This doesn't return a `Status` anymore, it returns a `Result<std::shared_ptr<arrow::json::TableReader>>`. The CSV docs reflect that by saving it to a variable with a `maybe_` prefix, so we could do the same here:
```suggestion
      auto maybe_reader = arrow::json::TableReader::Make(pool, input, read_options,
```